datastore/src/main/java/com/ds/datastore/RequestListener.java

```diff
@@ -21,6 +21,6 @@ public void requestInitialized(ServletRequestEvent sre){
             requestID = String.valueOf(UUID.randomUUID());
         }
         request.setAttribute("requestID", requestID);
-        logger.info("Call from {}, Request Type: {}, Request {} received", request.getHeader("referer"), request.getMethod(), requestID);
+        logger.info("Call from {}, Request Type: {}, Request {} received", request.getRequestURI(), request.getMethod(), requestID);
```

Comment:
one last thing: "Call `to`, not from"

Reply:
But the listener intercepts incoming requests and therefore request.getRequestURI() has the URI of the server that the call is coming from

Reply:
https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServletRequest.html

The request URI is the `uri` requested by the client. 

`getRemoteHost` would give you the client hostname - feel free to add it if you want.